### DISCUSSION TITLE
Picklist Drag&Drop Multiple Selection

### DISCUSSION THREAD

**Discussion by Mnaaz** [primefaces/community#1691] [PrimeNG]
The picklist currently allows the selection of multiple items via setting` [metaKeySelection]="true"`. However, when having set `[dragdrop]="true"` it would be nice to be able to move the multiple selection via drag&drop from source to target list and vice versa. This is currently not possible. Only the item that was selected first is moved.
This behaviour is also not consistent with being able to transfer multiple selection via the arrow buttons.
It would be great if this enhancement would be added and to have some documentation on how to achieve this via workaround until the feature is available.

**Comment by cetincakiroglu** [comment]
Hi,

Thanks for the feedback, we'll evaluate this in v18

**Comment by Mnaaz** [comment]
That's nice, thanx. Currently, we have implemented a workaround to select multiple items and drag&drop them. We do this by overwriting the array that is provided by onSourceSelect/onTargetSelect and onMoveToSource/onMoveToTarget but it seems that in some constellations, tracking is lost concerning the selected items. When selecting an item in the source, sometimes the array contains the wrong items for instance.
So, it would be cleaner to have the multiselect feature for drag&drop via PrimeNG itself.
